File: tasking/sources/task_dep.9.c
/*
* @@name:	task_dep.6c
* @@type:	C
* @@compilable:	yes
* @@linkable:	yes
* @@expect:	success
* @@version:	omp_5.0
*/
#include <stdio.h>
int main()
{
   int a, b, c, d;
   #pragma omp parallel
   #pragma omp single
   {
      #pragma omp task depend(out: c)
         c = 1;   /* Task T1 */
      #pragma omp task depend(out: a)
         a = 2;   /* Task T2 */
      #pragma omp task depend(out: b)
         b = 3;   /* Task T3 */
      #pragma omp task depend(in: a) depend(mutexinoutset: c)
         c += a;  /* Task T4 */
      #pragma omp task depend(in: b) depend(mutexinoutset: c)
         c += b;  /* Task T5 */ 
      #pragma omp task depend(in: c)
         d = c;   /* Task T6 */
   }
   printf("%d\n", d);
   return 0;
}

File: tasking/sources/task_dep.9.f90
! @@name:	task_dep.6f
! @@type:	F-free
! @@compilable:	yes
! @@linkable:	yes
! @@expect:	success
! @@version:    omp_5.0
program example
   integer :: a, b, c, d
   !$omp parallel
   !$omp single
      !$omp task depend(out: c)
      c = 1      ! Task T1
      !$omp end task
      !$omp task depend(out: a)
      a = 2      ! Task T2
      !$omp end task
      !$omp task depend(out: b)
      b = 3      ! Task T3
      !$omp end task
      !$omp task depend(in: a) depend(mutexinoutset: c)
      c = c + a  ! Task T4
      !$omp end task
      !$omp task depend(in: b) depend(mutexinoutset: c)
      c = c + b  ! Task T5
      !$omp end task
      !$omp task depend(in: c)
      d = c      ! Task T6
      !$omp end task
   !$omp end single
   !$omp end parallel
   print *, d
end program
